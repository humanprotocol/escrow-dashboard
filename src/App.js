import { ThemeProvider } from "@mui/material/styles";

import theme from "./theme";
import Main from "./components/Main";
import "./App.css";
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <Main />
      <ToastContainer
        position="top-center"
        autoClose={5000}
        hideProgressBar
        newestOnTop
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />
    </ThemeProvider>
  );
}

export default App;
