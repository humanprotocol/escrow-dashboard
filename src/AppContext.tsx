import React from "react";

export default React.createContext({
  network: "polygon",
  setNetwork: (value: string): void => {},
});
