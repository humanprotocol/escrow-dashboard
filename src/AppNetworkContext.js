import React from 'react';

export default React.createContext({
<<<<<<< HEAD:src/AppContext.tsx
  network: "polygon",
  setNetwork: (value: string): void => {},
=======
  network: 'polygon',
  setNetwork: () => null,
>>>>>>> master:src/AppNetworkContext.js
});
