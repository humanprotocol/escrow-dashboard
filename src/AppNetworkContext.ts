import * as React from 'react';

export default React.createContext({
  network: 'polygon',
  // eslint-disable-next-line no-unused-vars
  setNetwork: (value: string): void => {},
});
