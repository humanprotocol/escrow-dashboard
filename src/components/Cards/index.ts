export { CardTextBlock } from './CardTextBlock';
export { CardBlockWithChildren } from './CardBlockWithChildren';
export { CardLinkBox } from './CardLinkBox';
