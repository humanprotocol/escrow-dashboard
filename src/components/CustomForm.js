import React, {useState} from 'react';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Stack from "@mui/material/Stack";

const CustomForm = ({ status, message, onValidated }) => {
    const [email, setEmail] = useState('');

    const handleSubmit = (e) => {
      e.preventDefault();
      email &&
      email.indexOf("@") > -1 &&
      onValidated({
          EMAIL: email,
      });
    }

    return (
      <form className="mc__form" onSubmit={(e) => handleSubmit(e)}>
        {status === "sending" && (
          <div className="mc__alert mc__alert--sending" style={{marginBottom: '20px'}}>
            sending...
          </div>
        )}
        {status === "error" && (
          <div 
            className="mc__alert mc__alert--error"
            dangerouslySetInnerHTML={{ __html: message }}
            style={{marginBottom: '20px'}}
          />
        )}
        {status === "success" && (
          <div
            className="mc__alert mc__alert--success"
            dangerouslySetInnerHTML={{ __html: message }}
            style={{marginBottom: '20px'}}
          />
        )}

        <Stack
            direction={{ lg: "row", md: "row", sm: "row", xs: "column" }}
            spacing={2}
            sx={{
              width: "100%"
            }}
          >
          <div className="mc__field-container">
            <TextField
                variant="outlined"
                size="small"
                label="Email"
                sx={{width: "calc(100%)",  flexBasis: "100%"}}
                onChange={(e) => setEmail(e.target.value)}
              >
                {email}
            </TextField>
          </div>
            <Button
              type="submit"
              sx={{backgroundColor: '#320A8D', color: 'white', textTransform: 'capitalize', paddingLeft: '40px', paddingRight: '40px', marginLeft: '20px'}}
            >
                Subscribe
            </Button>
          </Stack>
      </form>
    );
};

export default CustomForm;