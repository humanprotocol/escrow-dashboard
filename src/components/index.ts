export { Escrow, EscrowFactoryView } from './Escrow';
export { App, AppNetworkContext } from './App';
export { NetworkSwitcher } from './NetworkSwitcher';
export { CardBlockWithChildren, CardTextBlock } from './Cards';
export { SkeletonLaunched } from './Skeletons';
export { Navbar } from './Navbar';
export { Header } from './Header';
export { Footer } from './Footer';
export { Layout } from './Layout';
export { Search } from './Search';
