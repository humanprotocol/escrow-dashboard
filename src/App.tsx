import * as React from "react";

import { ThemeProvider } from "@mui/material/styles";
import { ApolloProvider } from "@apollo/client";

import theme from "./theme";
import Main from "./components/Main";
import "./App.css";
import { getClient } from "./queries";
import { networkMap } from "./constants";
import AppContext from "./AppContext";


function App() {
  const [network, setNetwork] = React.useState<string>("polygon");

  return (
    <AppContext.Provider value={{ network, setNetwork }}>
      <ApolloProvider client={getClient(networkMap[network].graphqlClientUrl)}>
        <ThemeProvider theme={theme}>
          <Main />

        </ThemeProvider>
      </ApolloProvider>
    </AppContext.Provider>
  );
}

export default App;
