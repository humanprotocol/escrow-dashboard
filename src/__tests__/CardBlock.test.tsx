import * as React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { CardTextBlock } from 'src/components';

const mock = {
  value: 'Value',
  title: 'Title',
};

describe('when rendered CardTextBlock component', () => {
  it('should render passed prop value', () => {
    render(<CardTextBlock value={mock.value} title={mock.title} />);
    expect(screen.findByLabelText(mock.value)).toBeTruthy();
  });

  it('should render passed prop title', () => {
    render(<CardTextBlock value={mock.value} title={mock.title} />);
    expect(screen.findByLabelText(mock.title)).toBeTruthy();
  });
});

it('renders correctly', () => {
  const tree = renderer
    .create(<CardTextBlock value={mock.value} title={mock.title} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
