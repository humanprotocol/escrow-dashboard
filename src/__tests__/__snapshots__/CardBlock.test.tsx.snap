// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-n4pdsy-MuiDivider-root"
    role="separator"
  >
    <span
      className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Title
      </p>
    </span>
  </div>,
  <p
    className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
  >
    Value
  </p>,
]
`;
